### PR TITLE
Add a table of drop rates for every item

### PR DESCRIPTION
**Pull request by Zollo757347**
- Added a table of drop rates for every item, as the following image shows
- Because some crates may drop multiple loots, some items may have their calculated drop rates higher than the loot table

image

### CODE REVIEW

**Comment by kenos1** [comment]
Please format your pull request with `pnpm lint`

**Comment by kenos1** [comment]
It also appears that you added the same component for every article file. Try putting the table in all articles (page/layout.tsx).

**Comment by Zollo757347** [comment]
I think I can't really do that, because the table is now located at the middle of the article, under "Obtaining" section. Unlike sidebars which can be conveniently placed at the end of every article. I can also put the table at the end of the page though, but that will make it separated with "Obtaining" section, and I think that is a little bit counterintuitive.

**Comment by kenos1** [comment]
Fair point, I'm gonna ask @Compositr to review cuz idk how this API stuff works in NextJS

**Comment by Compositr** [comment]
If you want to insert the table in the middle of the article we'd need to rework how articles are rendered unfortunately.

Also, the old way we used to do with `layout.tsx` is deprecated in favour of the new single-file `page.tsx` method (see https://github.com/HasangerGames/suroi-wiki/blob/main/app/(wiki)/buildings/%5Bitem%5D/page.tsx)

**Comment by kenos1** [comment]
solidjs when

**Comment by Zollo757347** [comment]
Manually adding a little component for each article isn't that bad in my opinion (that's why I chose to do it), but of course it's up to you to decide whether to keep this or not. If you don't like how this is implemented, feel free to close the PR. I will purpose the same PR if the rework is finished in the future.

**Comment by Zollo757347** [comment]
I have an idea that we can move this functionality into a separate page. We can put a link to this page in the sidebars or in the loot table page, so it won't be isolated. If that's good, then I'll work on that asap.

**Comment by Compositr** [comment]
I think we should just add a link to the loot tables where the item appears in the sidebar sort of how we have links to where buildings appear, etc.

You can link to loot tables by passing their string ID as a `#` to the url (like wiki.suroi.io/#id string here)

**Comment by Zollo757347** [comment]
I agree with your point, but I may prefer adding links to the obstacles that drops the item, instead of the loot tables. If I click on [toilet](https://wiki.suroi.io/loot#toilet) and it leads me to the loot table, I'll be really confused. Also I'll add another link to the loot table on the obstacle sidebar so people can still get there if they want.

**Comment by Zollo757347** [comment]
Now it looks something like this. (Lewis Gun)
drawing
And this is how it looks like in obstacle sidebar.
drawing
I am planning to partially rewrite loot table page in another PR because some of the values are wrongly calculated, so I'll leave the api thing there.

**Review by Compositr — CHANGES_REQUESTED** [review]
I think the "Obstacles" section of the sidebar should be made more clear that it is a chance (ie, it uses a loot table). Maybe reword it to be like "Found In" or something similar.

**Comment by Compositr** [comment]
Thanks for your help by the way, we appreciate it. Are you in the Discord server? We could coordinate easier over there if you are willing.

And yes, the code for the loot tables is flawed in a few places although its such a mess I don't want to touch it lol. Its fairly accurate for smaller loot tables though.

**Comment by Zollo757347** [comment]
Yes "location" is not the perfect frase, but English is not my first language so I couldn't come up with a suitable phrase for that lol. 

And yes I am in the server. My username is @zollo757347 as well.

**Review by Compositr — APPROVED** [review]
.

**Comment by kenos1** [comment]
,

**Comment by Arman-AIi** [comment]
;

**Comment by 1092384** [comment]
:
